modules/navier_stokes/src/base/NavierStokesApp.C

```diff
@@ -53,34 +53,42 @@ associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
   // add variables action
   registerTask("add_navier_stokes_variables", /*is_required=*/false);
   addTaskDependency("add_navier_stokes_variables", "add_variable");
+  addTaskDependency("add_mortar_variable", "add_navier_stokes_variables");
```

Comment:
The old dependency resolver used to insert a new task immediately behind its prereq, and in front of other things that depend on the prereq ... but not always. In fact what it did was not very transparent at all. With the new dependency resolver you explicitly add every dependency edge; it's fully transparent and obvious what is happening. Now does `add_mortar_variable` truly depend on `add_navier_stokes_variables`? I would sure hope not! But where does the dependency appear? It is hard for me to say except through Monte Carlo. It should definitely happen before `add_user_object`; but should it happen before `add_fluid_properties` for example? I don't actually know unless I try and then maybe I get an error in the FP module. Maybe I don't, but then an error crops up in SAM. What I do know is that if I glance at the old sorting of tasks in the debugger, `add_navier_stokes_variables` occurred before `add_mortar_variable` and so I know this is a safe location for it in the graph.

Reply:
Interesting. This seems like there should be convention when there is indetermination on the order. Like "alphabetical sorting"
adding dependencies makes the graph larger but since your algorithm is faster we dont notive it

Reply:
but adding dependencies also makes the graph more consistent across apps, which is good